packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx

```diff
@@ -1,94 +1,82 @@
 
-const DrawerSection: React.FC<any> = ({
-  section,
-  open,
-  level = 0,
-  onSelect,
-}) => {
-  // Constants
-  const isMainSection = level === 0;
-  const isSelected =
-    section.isSelected ||
-    section.subSections?.some((subsection: Section) => subsection.isSelected);
-
-  // States
+const DrawerSection = ({ item, onClick }: DrawerSectionProps<Menu>) => {
+  const isMainSection = item.type === 'folder';
+  const isSelected = false;
   const [expanded, setExpanded] = useState(false);
 
-  // Handlers
-  const handleClick = () => {
-    if (section.subSections) {
+  const handleClick = useCallback(() => {
+    if (item.type === 'folder') {
       setExpanded(prev => !prev);
-    }
-
-    if (section.isSelected || (section.subSections && expanded)) {
-      onSelect(null);
     } else {
-      onSelect(section.id);
+      console.warn('DrawerSection: unexpected type');
     }
-  };
+  }, [item]);
 
-  return (
-    <Box
-      sx={{
-        ...styles.drawerSectionBox,
-        backgroundColor:
-          isMainSection && isSelected
-            ? theme.palette.baselineColor.neutral[10]
-            : 'transparent',
-        padding:
-          isMainSection && !!section.subSections && isSelected && '0.5rem',
-      }}>
-      <Box
-        onClick={handleClick}
-        sx={{
-          ...styles.listItemButton,
-          ...(section.isSelected && styles.listItemButtonSelected),
-          ...styles.listItemContentText,
-          borderRadius: open ? '0.5rem' : '12.5rem',
-          fontSize: open ? '1.5rem' : '1rem',
-          justifyContent: open ? 'space-between' : 'center',
-        }}>
-        <Box
-          sx={{
-            ...styles.listItemInnerContentText,
-            justifyContent: open ? 'flex-start' : 'center',
-          }}>
-          <Box sx={styles.listItemIconContent}>
-            <Typography sx={styles.listItemIconTypography}>
-              {section.icon}
-            </Typography>
-          </Box>
+  const mainStyle = useMemo(
+    () => ({
+      ...styles.drawerSectionBox,
+      backgroundColor:
+        isMainSection && isSelected
+          ? theme.palette.baselineColor.neutral[10]
+          : 'transparent',
+      padding:
+        isMainSection && !!item.submenu && isSelected ? '0.5rem' : undefined,
+    }),
+    [isMainSection, isSelected, item.submenu],
+  );
 
-          {open && (
-            <Typography sx={styles.listItemText}>{section.label}</Typography>
-          )}
-        </Box>
+  const titleContainerStyle = useMemo(
+    () => ({
+      ...styles.listItemButton,
+      ...(isSelected ? styles.listItemButtonSelected : undefined),
+      ...styles.listItemContentText,
+      borderRadius: '0.5rem',
+      fontSize: '1.5rem',
+      justifyContent: 'space-between',
+    }),
+    [isSelected],
+  );
+
+  const titleStyle = useMemo(
+    () => ({
+      ...styles.listItemInnerContentText,
+      justifyContent: 'flex-start',
+    }),
+    [],
+  );
```

Comment:
Using useMemo just to add a style does not seem to me a correct way to use it. Also by passing an empty dependency array, the value will only be calculated once when the component is mounted, which is the same as doing this:

const titleStyle = {
...styles.listItemInnerContentText,
      justifyContent: 'flex-start',
};